lib/tasks/package.rake

```diff
@@ -0,0 +1,109 @@
+require 'tempfile'
+require 'English'
+
+def check_git
+  modified = `git ls-files -m --exclude-standard .`
+  if modified.empty?
+    puts 'No uncommitted changes detected.'
+  else
+    raise "Warning: uncommitted changes!\n\n#{modified}\n"
+  end
+end
+
+def build_gem(package_dir)
+  Dir.chdir '..'
+  gemfilename = "suse-connect-#{SUSE::Connect::VERSION}.gem"
+
+  `rm suse-connect-*.gem` if Dir['*.gem'].any?
+  `gem build suse-connect.gemspec`
+
+  raise 'Gem build failed.' unless $CHILD_STATUS.exitstatus.zero?
+
+  sh "cp #{gemfilename} #{package_dir}"
+  puts "Gem built and copied to #{package_dir}." if $CHILD_STATUS.exitstatus.zero?
+end
+
+def checkout_package(obs_project, package_name, package_dir)
+  Dir.chdir package_dir
+  unless Dir['.osc'].any?
+    sh 'mkdir .tmp; mv * .tmp/'
+    sh "osc co #{obs_project} #{package_name} -o ."
+    sh 'mv .tmp/* .; rm -r .tmp/'
```

Comment:
I see that this stuff is from the task that we have currently in master.
However, I really don't like it: if you have any changes in OBS that you haven't checked out yet -- that would overwrite upstream changes with local changes, and then run `osc ar` in step 6. If you commit after that without examining `osc diff` -- those changes could be lost.

Reply:
I'm not sure I'm following entirely. This piece of code runs only if there has not been any checkout before. `SUSEConnect.spec` and SUSEConnect.changes` are also versioned with git, so if there are any local changes, like to the version number, we actually want them upstream in the end? The `.changes` file should never be touched manually anyways I think.

What I could implement is an `osc up` call, so files are aligned. Is this what you had in mind?

Reply:
I'm just saying it's possible that the working copy isn't up-to-date with the state on OBS. I personally wouldn't call neither `osc up`, nor `osc ar` and let the developer take care of that.

I think it would be cool to run `osc status` and examine it's output -- e.g., raise an error if the changes file hasn't been modified.